scripts/cnn_variant_wdl/cram2model.wdl

```diff
@@ -0,0 +1,431 @@
+    Int? cpu
+    command { 
+        java -Djava.io.tmpdir=tmp -jar ${gatk_override} \
+        HaplotypeCaller \
+        -R ${reference_fasta} \
+        -I ${input_bam} \
+        -O ${output_prefix}_hc4.vcf.gz \
+        -L ${interval_list} \
+        -bamout ${output_prefix}_bamout.bam \
+        ${extra_args}
+    }
+
+    output {
+        File bamout = "${output_prefix}_bamout.bam"
+        File bamout_index = "${output_prefix}_bamout.bai"
+        File raw_vcf = "${output_prefix}_hc4.vcf.gz"
+        File raw_vcf_index = "${output_prefix}_hc4.vcf.gz.tbi"
+    }
+    runtime {
+        docker: "broadinstitute/genomes-in-the-cloud:2.1.1"
+        memory: "3 GB"
+        disks: "local-disk " + disk_space_gb + " SSD"
+    }
+}
+
+task WriteTensors {
+    File input_bam
+    File input_bam_index
+    File input_vcf
+    File input_vcf_index
+    File reference_fasta
+    File reference_dict
+    File reference_fasta_index
+    File truth_vcf
+    File truth_vcf_index
+    File truth_bed
+    String output_prefix
+    String tensor_type
+    File interval_list
+
+    # Runtime parameters
+    File? gatk_override
+    Int? mem_gb
+    String gatk_docker
+    Int? preemptible_attempts
+    Int? disk_space_gb
+    Int? cpu
+
+    command {
+        mkdir "./tensors/"
+
+        java -Djava.io.tmpdir=tmp -jar ${gatk_override} \
+        CNNVariantWriteTensors \
+        -R ${reference_fasta} \
+        -V ${input_vcf} \
+        -truth-vcf ${truth_vcf} \
+        -truth-bed ${truth_bed} \
+        -tensor-type ${tensor_type} \
+        -output-tensor-dir "./tensors/" \
+        -bam-file ${input_bam}
+        
+        tar -czf "tensors.tar.gz" "./tensors/"
+    }
+
+    output {
+      File tensors = "tensors.tar.gz"
+    }
+    runtime {
+        docker: "samfriedman/p3"
+        memory: "3 GB"
+        disks: "local-disk " + disk_space_gb + " SSD"
+    }
+
+}
+
+
+task TrainModel {
+    Array[File] tar_tensors
+    String output_prefix
+    String tensor_type
+    Int epochs
+
+    # Runtime parameters
+    File? gatk_override
+    Int? mem_gb
+    Int? preemptible_attempts
+    Int? disk_space_gb
+    Int? cpu
+
+    command {
+        for tensors in ${sep=' ' tar_tensors}  ; do
+            tar -xzf $tensors 
+        done
+
+        java -Djava.io.tmpdir=tmp -jar ${gatk_override} \
+        CNNVariantTrain \
+        -input-tensor-dir "./tensors/" \
+        -model-name ${output_prefix} \
+        -image-dir "./" \
+        -tensor-type ${tensor_type} \
+        -epochs ${epochs}
+    }
+
+    output {
+        File model_json = "${output_prefix}.json"
+        File model_hd5 = "${output_prefix}.hd5"
+        File roc_png = "per_class_roc_${output_prefix}.png"
+        File training_png = "metric_history_${output_prefix}.png"
+    }
+
+    runtime {
+#        docker: "samfriedman/p3"
+#        memory: "16 GB"
+#        disks: "local-disk " + disk_space_gb + " HDD"
+
+      docker: "samfriedman/gpu"
+      gpuType: "nvidia-tesla-k80"
```

Comment:
Do these args work or is it a red herring like the minCpuPlatform?

Reply:
May need Pipelines API v2 to really work.  It kind of works now.

Reply:
Kind of works?  The way Elvis is kind of dead?  Either it works or it doesn't.